internal/fwserver/server_planresourcechange.go

```diff
@@ -142,7 +142,13 @@ func (s *Server) PlanResourceChange(ctx context.Context, req *PlanResourceChange
 	//
 	// We only do this if there's a plan to modify; otherwise, it
 	// represents a resource being deleted and there's no point.
-	if !resp.PlannedState.Raw.IsNull() && !resp.PlannedState.Raw.Equal(req.PriorState.Raw) {
+	//
+	// TODO: Need to verify this logic won't cause issues
+	//
+	// I added a new method to `terraform-plugin-go` that prevents the Equals method from panicking when there is a type mismatch.
+	// This code path wasn't possible to hit with framework's schema, but the introduction of dynamic types (which can potentially change types)
+	// has revealed this path.
+	if !resp.PlannedState.Raw.IsNull() && !resp.PlannedState.Raw.SafeEqual(req.PriorState.Raw) {
```

Comment:
Here is the linked draft PR: https://github.com/hashicorp/terraform-plugin-go/pull/383

Reply:
This fix has been applied upstream, will switch to using `v0.22.1` when it's released